File: projekt3/Dandelion.py
from Plant import *

class Dandelion(Plant):
    
    __DANDELION_STRENGTH__ = 0
    __COLOR__ = "Yellow"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__DANDELION_STRENGTH__, world)
    
    def getName(self):
        return "Dandelion"
    
    def getColor(self):
        return self.__COLOR__

    def spread(self, newRow, newCol):
        self._world_.setOrganism(newRow, newCol, Dandelion(newRow , newCol, self._world_))

    def takeAction(self):
        for i in range(3):
            random = randrange(100)
            if (random < 10):
                self.trySpread()
   

File: projekt2/src/proj/java/model/plant/Dandelion.java
package proj.java.model.plant;

import java.awt.Color;
import java.util.Random;

import proj.java.model.Plant;
import proj.java.model.World;

public class Dandelion extends Plant {

    private static final int DANDELION_STRENGTH = 0;

    @Override
    protected void spread(int newRow, int newCol, World world) {
        world.setOrganism(newRow, newCol, new Dandelion());
    }

    @Override
    public int getStrength() {
        return DANDELION_STRENGTH;
    }

    @Override
    public void takeAction(World world) {
        for (int i = 0; i < 3; i++) {
            var random = new Random();
            if (random.nextInt(100) < 10) {
                trySpread(world);
            }
        }

    }

    @Override
    public Color getColor() {
        return Color.YELLOW;
    }
}
